Assets/MixedRealityToolkit/_Core/Utilities/DebugUtilities.cs

```diff
@@ -1,56 +1,168 @@
-        public static void DebugAssert(bool condition, string message)
+        private static LoggingLevels MixedRealityToolkitLoggingLevel =>
+            (MixedRealityToolkit.Instance.ActiveProfile != null) ? MixedRealityToolkit.Instance.ActiveProfile.LoggingLevel : (LoggingLevels)(-1);
+
+        /// <summary>
+        /// Assert a condition and log a message if the condition is not satisfied, using the Mixed Reality Toolkit
+        /// object's currently configured logging level.
+        /// </summary>
+        /// <param name="condition">The condition to be evaluated.</param>
+        public static void Assert(bool condition)
+        {
+            Assert(MixedRealityToolkitLoggingLevel, condition);
+        }
+
+        /// <summary>
+        /// Assert a condition and log a message if the condition is not satisfied.
+        /// </summary>
+        /// <param name="logLevel">The logging level used to filter debug output.</param>
+        /// <param name="condition">The condition to be evaluated.</param>
+        public static void Assert(LoggingLevels logLevel, bool condition)
+        {
+            Assert(logLevel, condition, string.Empty);
+        }
+
+        /// <summary>
+        /// Assert a condition and log a message if the condition is not satisfied, using the Mixed Reality Toolkit
+        /// object's currently configured logging level.
+        /// </summary>
+        /// <param name="condition">The condition to be evaluated.</param>
+        /// <param name="message">Message to display if the condition is not satisfied.</param>
+        public static void Assert(bool condition, string message)
+        {
+            Assert(MixedRealityToolkitLoggingLevel, condition, message);
+        }
+
+        /// <summary>
+        /// Assert a condition and log a message if the condition is not satisfied.
+        /// </summary>
+        /// <param name="logLevel">The logging level used to filter debug output.</param>
+        /// <param name="condition">The condition to be evaluated.</param>
+        /// <param name="message">Message to display if the condition is not satisfied.</param>
+        public static void Assert(LoggingLevels logLevel, bool condition, string message)
+        {
+            if ((logLevel & LoggingLevels.Assert) != 0)
+            {
+                Debug.Assert(condition, message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a critial error message, using the Mixed Reality Toolkit
+        /// object's currently configured logging level.
+        /// </summary>
+        /// <param name="message">The message to display.</param>
+        /// <remarks>
+        /// To help differentiate the output of this method from that of LogError, the
+        /// string "Critical: " is prepended to the supplied message.
+        /// </remarks>
+        public static void LogCriticalError(string message)
+        {
+//            LogCriticalError(MixedRealityToolkitLoggingLevel, message);
```

Comment:
Why is this commented out?

Reply:
I was testing something and missed uncommenting it... thanks for the catch!